### ISSUE TITLE
Pressing delete at the beginning of a block (and NO merge) moves the focus to the wrong block

### ISSUE THREAD

**Issue by daniloercoli** [wordpress-mobile/gutenberg-mobile#306]
Pressing delete at the beginning of a block can move the focus to the wrong block when the merge of the 2 blocks cannot happen due to different ("unmergeable") block types.

![delete-focus-wrong](https://user-images.githubusercontent.com/518232/49300643-08e7a900-f4c3-11e8-8dd8-7014e1c036ee.gif)

To reproduce this issue:
- Start the demo app
- Move the cursor at the beginning of the Para block `Lorem...`
- Tap `delete` 
- You will the top "Page Break" block being selected, but the caret is still in the current block and the keyboard up

Note that in GB the standard behavior is to move the focus to the previous block if it's not mergeable. https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/store/effects.js#L165

This is not a problem of the onMerge code, but rather of the focus handling, since i can also replicate this with other steps. Link to the relevant ticket soon to arrive.

**Comment by koke** [comment]
Is this one still valid after #308 was closed? What should be the desired behavior?
Also, if this is for the beta (it's in that milestone) let's take it out of the list on #196 which is for the alpha.

**Comment by daniloercoli** [comment]
Yes, just tested and the issue is still there, even though the behavior is slightly different.
Will close this and open a new issue for the beta.
